core/src/main/java/org/apache/iceberg/BaseReplacePartitions.java

```diff
@@ -53,6 +61,45 @@ public ReplacePartitions validateAppendOnly() {
     return this;
   }
 
+  @Override
+  public ReplacePartitions validateFromSnapshot(long snapshotId) {
+    this.startingSnapshotId = snapshotId;
+    return this;
+  }
+
+  @Override
+  public ReplacePartitions validateNoConflictingAppends() {
+    this.validateNoConflictingAppends = true;
+    return this;
+  }
+
+  @Override
+  public void validate(TableMetadata currentMetadata) {
+    if (validateNoConflictingAppends) {
+      Expression conflictDetectionFilter;
+      if (writeSpec().isUnpartitioned()) {
+        // Unpartitioned table, check against all files
+        conflictDetectionFilter = Expressions.alwaysTrue();
+      } else {
+        conflictDetectionFilter = deletedPartitions.stream().map(p -> {
```

Comment:
I don't think it's necessary to convert the partition set to a filter. You should be able to just omit the filter when configuring the `ManifestGroup` and apply it to each data file returned by the group.

Reply:
Done